services/user.js

```diff
@@ -741,6 +748,57 @@ class UserService {
         });
         return await this.userCollection.aggregate(pipeline);
     }
+
+    #validateUserPermission(permissions, notifications) {
+        // Only Valid User Permissions
+        if (permissions) {
+            const allPermissionNames = [...Object.values(SUBMISSION_REQUEST), ...Object.values(DATA_SUBMISSION), ...Object.values(ADMIN)];
+            const allPermissionNamesSet = new Set(allPermissionNames);
```

Comment:
allPermissionNameSet and allEmailNotificationNamesSet are based on constants so they can be defined at initialization instead of each time this function is called

Reply:
Update it for the change.